src/main/java/de/tum/cit/aet/artemis/exam/service/StudentExamService.java

```diff
@@ -622,6 +622,10 @@ public void setUpTestExamExerciseParticipationsAndSubmissions(StudentExam studen
         List<StudentParticipation> generatedParticipations = Collections.synchronizedList(new ArrayList<>());
         setUpExerciseParticipationsAndSubmissions(studentExam, generatedParticipations);
         // TODO: Michael Allgaier: schedule a lock operation for all involved student repositories of this student exam (test exam) at the end of the individual working time
+        if (!generatedParticipations.isEmpty()) {
+            studentExam.setStudentParticipations(generatedParticipations);
```

Comment:
I think a comment explaining that this is required due to the multiple participations in test exams might be beneficial

Reply:
Thanks! 🙏